### PR TITLE
Feature/BDN-511: Update Vungle SDK to 7.0.0

### PR DESCRIPTION
**Pull request by AntonZarytski**
https://appodeal.atlassian.net/browse/BDN-511

### CODE REVIEW

File: adapter/vungle/src/main/java/org/bidon/vungle/VungleAdapter.kt [hunk 1]
```diff
@@ -73,24 +71,12 @@ class VungleAdapter :
...
-            } else {
-                Vungle.Consent.OPTED_OUT
-            }
-            Vungle.updateConsentStatus(status, null)
-        }
-        if (regulation.coppaApplies) {
-            Vungle.updateUserCoppaStatus(true)
+            VunglePrivacySettings.setGDPRStatus(regulation.hasGdprConsent, null)
         }
+        VunglePrivacySettings.setCOPPAStatus(regulation.coppaApplies)
```
Comment (klukwist) [inline adapter/vungle/src/main/java/org/bidon/vungle/VungleAdapter.kt]: Please set COPPA only if it applies 
`if (regulation.coppaApplies) {....`

File: adapter/vungle/src/main/java/org/bidon/vungle/VungleBannerAuctionParams.kt [hunk 2]
```diff
@@ -18,19 +17,15 @@ class VungleBannerAuctionParams(
...
+            BannerFormat.Adaptive -> if (isTablet) BannerAdSize.BANNER_LEADERBOARD else BannerAdSize.BANNER
         }
-    val config by lazy {
-        BannerAdConfig().apply {
-            this.adSize = bannerSize
-        }
-    }
 }
 
 data class VungleFullscreenAuctionParams(
```
Comment (klukwist) [inline adapter/vungle/src/main/java/org/bidon/vungle/VungleBannerAuctionParams.kt]: Remove `data` as it has now `val activity: Activity` which does not comply to data class.

File: adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleBannerImpl.kt [hunk 3]
```diff
@@ -75,82 +71,69 @@ internal class VungleBannerImpl :
...
-                                ad = ad,
-                                adValue = AdValue(
-                                    adRevenue = adParam.price / 1000.0,
-                                    precision = Precision.Precise,
-                                    currency = AdValue.USD,
-                                )
-                            )
+        val adListener = object : BaseAdListener {
+            override fun onAdLoaded(baseAd: BaseAd) {
+                logInfo(TAG, "onAdLoad =${baseAd.placementId}. $this")
```
Comment (klukwist) [inline adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleBannerImpl.kt]: It seems emitting event should be here
```
emitEvent(AdEvent.Fill(bidonAd))
```

File: adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleBannerImpl.kt [hunk 4]
```diff
@@ -75,82 +71,69 @@ internal class VungleBannerImpl :
...
-                        emitEvent(
-                            AdEvent.PaidRevenue(
-                                ad = ad,
-                                adValue = AdValue(
-                                    adRevenue = adParam.price / 1000.0,
-                                    precision = Precision.Precise,
-                                    currency = AdValue.USD,
-                                )
-                            )
+        val adListener = object : BaseAdListener {
```
Comment (klukwist) [inline adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleBannerImpl.kt]: I suppose it better to move `adListener` into `activity.runOnUiThread`, doesn't it?

File: adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleBannerImpl.kt [hunk 5]
```diff
@@ -75,82 +71,69 @@ internal class VungleBannerImpl :
...
+                    context = adParams.activity,
+                    placementId = adParams.bannerId,
+                    adSize = BannerAdSize.BANNER
+                ).apply {
+                    this.adListener = adListener
+                    load()
                 }
-            ).also {
-                it?.disableLifeCycleManagement(true)
+                emitEvent(AdEvent.Fill(bidonAd))
```
Comment (klukwist) [inline adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleBannerImpl.kt]: Wrong place for emitting `AdEvent.Fill`, cause we have just started loading here.

File: adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleInterstitialImpl.kt [hunk 6]
```diff
@@ -35,20 +36,74 @@ internal class VungleInterstitialImpl :
     StatisticsCollector by StatisticsCollectorImpl() {
 
     private var adParams: VungleFullscreenAuctionParams? = null
+    private var interstitialAd: InterstitialAd? = null
+    private val interstitialListener = object : BaseAdListener {
```
Comment (klukwist) [inline adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleInterstitialImpl.kt]: let's create listener into `override fun load(adParams: VungleFullscreenAuctionParams) {` and don't keep the instance in the class.

File: adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleRewardedImpl.kt [hunk 7]
```diff
@@ -35,20 +36,79 @@ internal class VungleRewardedImpl :
     StatisticsCollector by StatisticsCollectorImpl() {
 
     private var adParams: VungleFullscreenAuctionParams? = null
+    private var rewardedAd: RewardedAd? = null
+    private var rewardedListener = object : RewardedAdListener {
```
Comment (klukwist) [inline adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleRewardedImpl.kt]: the same as interstitial

File: adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleRewardedImpl.kt [hunk 8]
```diff
@@ -62,89 +122,24 @@ internal class VungleRewardedImpl :
 
     override fun load(adParams: VungleFullscreenAuctionParams) {
         this.adParams = adParams
```
Comment (klukwist) [inline adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleRewardedImpl.kt]: please, don't keep the whole `adParams` as a local class variable, cause it has Activity-reference. You need the only placement and payload.

**Review by klukwist — CHANGES_REQUESTED** [review]

**Comment by klukwist** [comment]
+ ktlint

File: adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleBannerImpl.kt (hunk 4 above)
Comment (AntonZarytski) [reply adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleBannerImpl.kt]: I don't think so, I'm not in favour of doing any work at UI. 
As far as I can see, you don't use UI thread further on the stack, so there is no point in creating a listener in UI thread.

File: adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleRewardedImpl.kt (hunk 8 above)
Comment (AntonZarytski) [reply adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleRewardedImpl.kt]: I understand correctly that I can remove of adParams in all adType classes?

Comment (klukwist) [reply adapter/vungle/src/main/java/org/bidon/vungle/impl/VungleRewardedImpl.kt]: I don't think so (if you talk about other Adapters). I removed all adParams with context. If you see adParams with Context in other adapter, please, inform me.

**Review by klukwist — APPROVED** [review]

**Pull request merged**